### PR TITLE
Removed Inheritance in ARR

### PR DESCRIPTION
**Pull request by NavGill12**
I removed Inheritance in ARR, instead, I am making use of the Interface object to access the update quantity() method, I realised it would help implement the Liskov Substitution Principle better.

### CODE REVIEW

File: src/main/java/entities/Item.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public Item(String[] itemInfo) throws ParseException {
             if(itemInfo[5].equals("N/A")){
                 this.expirationDates = null;}
             else{
-                DateFormat dateFormat = new SimpleDateFormat("MM-dd-yyyy");
+                DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
                 this.expirationDates = dateFormat.parse(itemInfo[5]);}
```
Comment (Robinrswanson) [inline src/main/java/entities/Item.java]: If possible I would like to keep the original format as that is how the dates are stored currently. Please let me know if necessary to keep I can change.

**Review by Robinrswanson — COMMENTED** [review]
Good work! Just let me know about the date formmating.

File: src/main/java/entities/Item.java (hunk 1 above)
Comment (NavGill12) [reply src/main/java/entities/Item.java]: For some reason, it was throwing a format error, I'm not sure but I think the sample data had dates stored in MM/dd/yyyy format.

Comment (NavGill12) [reply src/main/java/entities/Item.java]: I just realised, may be I hadn't updated my sample data, my bad, I'll change that!!

**Comment by tonyxieuoft** [comment]
The tests look good, now we have full test coverage!

**Pull request merged**